### PR TITLE
Add: filetype - Infer file type checking the magic numbers 

### PR DESCRIPTION
**Pull request by h2non**

### CODE REVIEW

**Comment by shurcooL** [comment]
Package looks good, but would you mind squashing this into a single commit and removing the unnecessary trailing whitespace?

**Comment by h2non** [comment]
@shurcooL what about now?

**Comment by shurcooL** [comment]
That's better, but it's still 2 commits. Would you like to learn how to do this (it'll be helpful elsewhere too), or would you prefer not to? The git history of this repo is not that important, but it helps to keep it a _little_ more tidy.

**Comment by h2non** [comment]
Oh! That was my fault. Sorry about that. Now it's strictly clean and pure.

**Comment by shurcooL** [comment]
Great. Now it's one commit, but there's a space in the link, causing it to broken:

```
[filetype](https://g ithub.com/h2non/filetype)
```

**Comment by h2non** [comment]
Oh god, this is the most silly PR I ever done for a while. Thank you for the patience.

**Comment by shurcooL** [comment]
No problem. LGTM.

File: README.md [hunk 1]
```diff
@@ -752,6 +752,7 @@ See [go-hardware](https://github.com/rakyll/go-hardware) for a comprehensive lis
 * [coop](https://github.com/rakyll/coop) - Cheat sheet for some of the common concurrent flows in Go.
 * [delve](https://github.com/derekparker/delve) - Go debugger.
 * [fastlz](https://github.com/fromYukki/fastlz) - Wrap over [FastLz](http://fastlz.org/) (free, open-source, portable real-time compression library) for GoLang.
+* [filetype](https://github.com/h2non/filetype) - Small package to Infer the file type checking the magic numbers signature.
```
Comment (shurcooL) [inline README.md]: Really minor, but is Infer meant to be capitalized in the middle of the sentence? :P

**Comment by h2non** [comment]
Are you challenging me, eh? :P 
There you go the fix, hopefully the final one ;)

**Comment by shurcooL** [comment]
Thanks!

**Comment by shurcooL** [comment]
> Are you challenging me, eh? :P

Nope, just pointing out a possible way to make your change slightly better. :)

**Pull request merged**